Use sysconf(_SC_NPROCESSORS_ONLN) to properly omit disabled smt cores.

Cap the maximum number of detected cores to 4; on many-cpu systems, kernel
locking takes significant cpu time. This could be revised as SMT in OpenBSD
improves, and the value can be overridden in /etc/gitconfig if needed e.g.

[grep]
	threads = 8
[index]
	threads = 8
[pack]
	threads = 8

Index: thread-utils.c
--- thread-utils.c.orig
+++ thread-utils.c
@@ -24,10 +24,12 @@ int online_cpus(void)
 	return 1;
 #else
 #ifdef _SC_NPROCESSORS_ONLN
+#define     MIN(a,b) (((a)<(b))?(a):(b))
 	long ncpus;
-#endif
 
-#ifdef GIT_WINDOWS_NATIVE
+	if ((ncpus = (long)sysconf(_SC_NPROCESSORS_ONLN)) > 0)
+		return MIN((int)ncpus, 4);
+#elif defined(GIT_WINDOWS_NATIVE)
 	SYSTEM_INFO info;
 	GetSystemInfo(&info);
 
@@ -55,11 +57,6 @@ int online_cpus(void)
 	if (!sysctl(mib, 2, &cpucount, &len, NULL, 0))
 		return cpucount;
 #endif /* defined(HAVE_BSD_SYSCTL) && defined(HW_NCPU) */
-
-#ifdef _SC_NPROCESSORS_ONLN
-	if ((ncpus = (long)sysconf(_SC_NPROCESSORS_ONLN)) > 0)
-		return (int)ncpus;
-#endif
 
 	return 1;
 #endif
